fvm/meter/weighted/meter.go

```diff
@@ -0,0 +1,193 @@
+package weighted
+
+import (
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/fvm/errors"
+	interfaceMeter "github.com/onflow/flow-go/fvm/meter"
+)
+
+// MeterInternalPrecisionBytes are the amount of bytes that are used internally by the Meter
+// to allow for metering computation smaller than one unit of computation. This allows for more fine weights.
+// A weight of 1 unit of computation is equal to 1<<16. The minimum possible weight is 1/65536.
+const MeterInternalPrecisionBytes = 16
+
+var (
+	// DefaultComputationWeights is the default weights for computation intensities
+	// these weighs make the computation metering the same as it was before dynamic execution fees
+	DefaultComputationWeights = map[uint]uint64{
+		uint(common.ComputationKindStatement):          1 << MeterInternalPrecisionBytes,
+		uint(common.ComputationKindLoop):               1 << MeterInternalPrecisionBytes,
+		uint(common.ComputationKindFunctionInvocation): 1 << MeterInternalPrecisionBytes,
+	}
+	// DefaultMemoryWeights are empty as memory metering is not fully defined yet
+	DefaultMemoryWeights = map[uint]uint64{}
+)
+
+var _ interfaceMeter.Meter = &Meter{}
+
+// Meter collects memory and computation usage and enforces limits
+// for any each memory/computation usage call it sums intensity multiplied by the weight of the intensity to the total
+// memory/computation usage metrics and returns error if limits are not met.
+type Meter struct {
+	computationUsed  uint64
+	computationLimit uint64
+	memoryUsed       uint64
+	memoryLimit      uint64
+
+	computationIntensities map[uint]uint
+	memoryIntensities      map[uint]uint
```

Comment:
are these fields supposed to be a 1:1 mapping from `basic.Meter`? If so, maybe possible to include `basic.Meter` type as a type-inclusion/field?

But having said that, if there are no common logic/code/methods to re-use, then the current approach might be better.

Reply:
the basic one is intended for testing purposes, so I think its ok to keep them separate for now.